emission/analysis/section_features.py

```diff
@@ -154,13 +176,37 @@ def calAvgSpeed(segment):
 # segment0: speed0 / (t1 - t0), segment1: (speed1 - speed0)/(t2-t1),
 # segment2: (speed2 - speed1) / (t3-t2)
 
+# def calAccels(segment):
+#   from dateutil import parser
+
+#   speeds = calSpeeds(segment)
+#   trackpoints = segment['track_points']
+
+#   if speeds is None or len(speeds) == 0:
+#     return None
+
+#   accel = np.zeros(len(speeds) - 1)
+#   prevSpeed = 0
+#   for (i, speed) in enumerate(speeds[0:-1]):
+#     currSpeed = speed # speed0
+#     speedDelta = currSpeed - prevSpeed # (speed0 - 0)
+#     # t1 - t0
+#     timeDelta = parser.parse(trackpoints[i+1]['time']) - parser.parse(trackpoints[i]['time'])
+#     # logging.debug("while calculating accels from %s -> %s, speedDelta = %s, timeDelta = %s" %
+#     #   (trackpoints[i+1], trackpoints[i], speedDelta, timeDelta))
+#     if timeDelta.total_seconds() != 0:
+#       accel[i] = speedDelta/(timeDelta.total_seconds())
+#       # logging.debug("resulting acceleration is %s" % accel[i])
+#     prevSpeed = currSpeed
+#   return accel
+
 def calAccels(segment):
   from dateutil import parser
 
-  speeds = calSpeeds(segment)
-  trackpoints = segment['track_points']
+  speed = calSpeed(segment)
```

Comment:
you can't replace `calSpeeds` by `calSpeed` because `calSpeeds` computes the speeds between each pair of points along the section. Note that this is already present in the new cleaned section object as `speeds`